src/gei/Commands/MigrateArchiveRepoCommand.cs

```diff
@@ -0,0 +1,194 @@
+                }
+            }
+
+            if (noSslVerify)
+            {
+                _log.LogInformation("SSL verification disabled");
+            }
+
+            var githubApi = noSslVerify ? _sourceGithubApiFactory.CreateClientNoSSL(ghesApiUrl) : _sourceGithubApiFactory.Create(ghesApiUrl);
+            var azureApi = noSslVerify ? _azureApiFactory.CreateClientNoSSL(azureStorageConnectionString) : _azureApiFactory.Create(azureStorageConnectionString);
+
+            var gitDataArchiveId = await githubApi.StartGitArchiveGeneration(githubSourceOrg, sourceRepo);
+            _log.LogInformation($"Archive generation of git data started with id: {gitDataArchiveId}");
+            var metadataArchiveId = await githubApi.StartMetadataArchiveGeneration(githubSourceOrg, sourceRepo);
+            _log.LogInformation($"Archive generation of metadata started with id: {metadataArchiveId}");
+
+            var metadataArchiveUrl = await WaitForArchiveGeneration(githubApi, githubSourceOrg, metadataArchiveId);
+            _log.LogInformation($"Archive (metadata) download url: {metadataArchiveUrl}");
+            var gitArchiveUrl = await WaitForArchiveGeneration(githubApi, githubSourceOrg, gitDataArchiveId);
+            _log.LogInformation($"Archive (git) download url: {gitArchiveUrl}");
+
+            var timeNow = $"{dateTimeNow:yyyy-MM-dd_HH-mm-ss}";
+
+            var gitArchiveFileName = $"{timeNow}-{Git_Archive_File_Name}";
+            var metadataArchiveFileName = $"{timeNow}-{Metadata_Archive_File_Name}";
+
+            var gitArchiveFilePath = $"/tmp/{gitArchiveFileName}";
+            var metadataArchiveFilePath = $"/tmp/{metadataArchiveFileName}";
+
+            // Download both archives to the local filesystem
+            _log.LogInformation($"Downloading archive from {gitArchiveUrl} to {gitArchiveFilePath}");
+            await azureApi.DownloadFileTo(gitArchiveUrl, gitArchiveFilePath);
+            _log.LogInformation($"Downloading archive from {metadataArchiveUrl} to {metadataArchiveFilePath}");
+            await azureApi.DownloadFileTo(metadataArchiveUrl, metadataArchiveFilePath);
+
+            _log.LogInformation($"Uploading archive {gitArchiveFileName} to Azure Blob Storage");
+            var authenticatedGitArchiveUrl = await azureApi.UploadToBlob(gitArchiveFileName, gitArchiveFilePath);
+            _log.LogInformation($"Uploading archive {metadataArchiveFileName} to Azure Blob Storage");
+            var authenticatedMetadataArchiveUrl = await azureApi.UploadToBlob(metadataArchiveFileName, metadataArchiveFilePath);
+
+            _log.LogInformation($"Deleting local archive files");
+            File.Delete(gitArchiveFilePath);
+            File.Delete(metadataArchiveFilePath);
+
+            // Run migrate repo command
+            var migrateRepoCommand = new MigrateRepoCommand(_log, _targetGithubApiFactory, _environmentVariableProvider);
```

Comment:
We should inject this dependency here.

Reply:
Updated in https://github.com/github/gh-gei/pull/221/commits/f92cebdab5a90a2a073fe05014295d4d64c10c3b